pushapkscript/task.py

```diff
@@ -8,15 +8,49 @@
 
 
 def extract_android_product_from_scopes(context):
-    prefix = context.config['taskcluster_scope_prefix']
+    prefixes = _get_scope_prefixes(context)
+    scopes = _extract_scopes_from_unique_prefix(
+        scopes=context.task['scopes'],
+        prefixes=prefixes
+    )
     scope = get_single_item_from_sequence(
-        context.task['scopes'],
-        condition=lambda scope: scope.startswith(prefix),
+        scopes,
+        condition=lambda _: True,     # scopes must just contain 1 single item
         ErrorClass=TaskVerificationError,
-        no_item_error_message='No valid scope found. Task must have a scope that starts with "{}"'.format(prefix),
-        too_many_item_error_message='More than one valid scope given',
+        no_item_error_message='No scope starting with any of these prefixes {} found'.format(prefixes),
+        too_many_item_error_message='More than one scope found',
     )
 
-    android_product = scope[len(prefix):]
+    android_product = scope.split(':')[-1]
 
     return android_product
+
+
+def _get_scope_prefixes(context):
+    prefixes = context.config['taskcluster_scope_prefixes']
+    return [
+        prefix if prefix.endswith(':') else '{}:'.format(prefix)
```

Comment:
Good thinking! I remember adding this auto-correction because there was discrepancy among the workers. The real fix is probably to not sanitize it here. 

Let's fix this in a followup.

Reply:
#56